### PR TITLE
fix(52517): JSDoc Comment Template not provided when unrelated JSDoc with tag exists

### LINKED ISSUE #52517: JSDoc Comment Template not provided when unrelated JSDoc with tag exists
JSDoc comment templates don't work if you have a preceding JSDoc comment declared using a `@typedef` or an `@overload` tag.

```ts
/** @typedef {string} Id */

/**$ */
function foo(x, y, z) {
}
```

Request `docCommentTemplate`.

The following is expected:

```js
/** @typedef {string} Id */

/**
 * 
 * @param x 
 * @param y 
 * @param z 
 */
function foo(x, y, z) {
}
```

But instead we get:

```js
/** @typedef {string} Id */

/**
 * 
 */
function foo(x, y, z) {
}
```

_Originally posted by @DanielRosenwasser in https://github.com/microsoft/TypeScript/issues/52370#issuecomment-1409159563_

### PR DESCRIPTION
**Pull request by a-tarasyuk**
Fixes #52517

### CODE REVIEW

File: tests/cases/fourslash/docCommentTemplateWithMultipleJSDocAndParameters.ts [hunk 1]
```diff
@@ -9,4 +9,8 @@
 /////*/**/
 ////function foo(p) {}
 
-verify.docCommentTemplateAt("", 3, "/** */");
```
Comment (a-tarasyuk) [inline tests/cases/fourslash/docCommentTemplateWithMultipleJSDocAndParameters.ts:12]: I think, in this case, we need to provide a template because the JSDoc comment that contains tags is not related to the host it is attached to. Or am I wrong? @DanielRosenwasser What do you think?

Comment (sandersn) [reply tests/cases/fourslash/docCommentTemplateWithMultipleJSDocAndParameters.ts:12]: That sounds right to me because of the intervening `/** */`.

However, you should also test the case without an intervening empty comment:

```
/** @param p */
/*/**/
function foo(p) { }
```

The seems like code that should not generate the template, but the current code might. I'm not sure.
(It's also possible that starting a fresh comment below the existing one implies that the author *always* wants a fresh template, but I think it's against the previous intent of the code in jsDoc.ts.)

**Review by sandersn — CHANGES_REQUESTED** [review]
Need one more test without an intervening empty comment.

File: tests/cases/fourslash/docCommentTemplateWithMultipleJSDoc3.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+/// <reference path='fourslash.ts' />
+
+/////** @param p */
+/////*/**/
```
Comment (sandersn) [inline tests/cases/fourslash/docCommentTemplateWithMultipleJSDoc3.ts:4]: I don't think this position should generate a template, because the comment on line 3 should apply to `foo` already. At least, I think that's the intent of the code that was previously in jsDoc.ts. @DanielRosenwasser might have a different opinion.

**Review by sandersn — CHANGES_REQUESTED** [review]

File: tests/cases/fourslash/docCommentTemplateWithMultipleJSDoc3.ts (hunk 2 above)
Comment (a-tarasyuk) [reply tests/cases/fourslash/docCommentTemplateWithMultipleJSDoc3.ts:4]: Sorry for the confusion caused by my misinterpretation of your previous comment.

**Review by sandersn — APPROVED** [review]
